### DISCUSSION TITLE
Steps to debug a deployment?

### DISCUSSION THREAD

**Discussion by james-woo** [vercel/vercel#4012] [General] [answered]
Hi all, I am having difficulties determining where my app is going wrong. My next js app works locally (`next dev`), but when I deploy it to Now, I am receiving 500 errors and I am not really sure how to debug the deployment (Github PR preview). What are the first things I should do?

Next JS 9.3.2
Firebase 7.13.1
React 16.13.1

Set up secrets with `now` CLI and set up `now.json`

I have a feeling that the environment variables are not being set correctly, but I am not sure how to get a console log. Any tips?

EDIT:
In the end it was a configuration issue, I was missing an environment variable in my `now.json`. I hope there is a way where I can get some logging to show up in a deployment version, or something that will help me debug 500 internal server errors.

**Comment by paulogdm** [comment, marked as answer]
You need to do the following:

- Check the [runtime logs](https://zeit.co/blog/refined-logging) of your app.
- Try to isolate the code misbehaving.
- Make a minimal reproducible example and simpler PoCs.

If you can share the minimal reproducible example, that would help us.

**Reply by james-woo** [reply]
I looked at the link for runtime logs, but it only shows build logs and serverless function logs. Is there any way I can get simple console logs to show?

**Reply by mcsdevv** [reply]
Hey @james-woo, the realtime logs will show any 500 errors from Serverless Functions, any application logging will be in the browser. If you're still having difficulty with this please provide a reproducible example and we'll take a look 😀
